lib/models/identifiers.ts

```diff
@@ -107,8 +118,28 @@ export namespace Identifiers {
         }
     }
 
+    export class CustomAppIdentifier {
+        constructor(
+            public identifier: CustomAppIdentifierEnum,
+            public value: string
+        ) {}
+
+        getParamValue(): string {
+            if (this.identifier === CustomAppIdentifierEnum.InternalId) {
```

Comment:
nitpick: Ts-pattern would be sexier here :P (no need for rework)

Reply:
100%, I'm looking forward to refactoring all of this and using `ts-pattern` :D